src/libmongoc/tests/unified/operation.c

```diff
@@ -1378,35 +1378,49 @@ operation_create_index (test_t *test,
    bson_parser_t *bp = NULL;
    char *name = NULL;
    bson_t *keys = NULL;
-   bson_t *create_indexes = NULL;
+   bool *unique = NULL;
+   bson_t *create_indexes = bson_new ();
    bson_t op_reply = BSON_INITIALIZER;
    bson_error_t op_error = {0};
    bson_t *opts = bson_new ();
+   bson_t *arguments = bson_new ();
+   bson_t *array = bson_new ();
 
```

Comment:
```suggestion
   bson_t arguments;
   bson_t array;
```

A `bson_t` may be heap allocated with `bson_new()` or stack allocated. Use stack allocated `bson_t` for a child `bson_t`.

If the `bson_t` used as a child in `BSON_APPEND_DOCUMENT_BEGIN` is heap allocated, it is leaked. This is a quirk of libbson.